amundsen_application/static/js/components/common/ResourceListItem/DashboardListItem/index.tsx

```diff
@@ -20,34 +26,42 @@ class DashboardListItem extends React.Component<DashboardListItemProps, {}> {
 
   getLink = () => {
     const { dashboard, logging } = this.props;
-    // TODO ttannis: Update when we know url
-    return `/dashboard`;
+    return `/dashboard/${dashboard.uri}?index=${logging.index}&source=${logging.source}`;
```

Comment:
Note: After taking a second look at the code I realized you intended the url it to be:
`/dashboard?uri=${dashboard.uri}&index=${logging.index}&source=${logging.source}` 
so that the `uri` is pulled form the search params on `DashboardPage`. 

I have this updated in an upcoming PR.